File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    int solve(int i, int x, int n, vector<int>&nums, vector<vector<int>>&dp){
        if(i == n) return x;
        if(dp[i][x] != -1) return dp[i][x];
        int pick = solve(i+1, x^nums[i], n, nums, dp);
        int notPick = solve(i+1, x, n, nums, dp);
        return dp[i][x] = pick + notPick;
    }

    int subsetXORSum(vector<int>& nums) {
        int n = nums.size();
        int x = 0;
        // int sum = 0;
        vector<vector<int>>dp(n, vector<int>(100, -1));
        return solve(0, x, n, nums, dp);
    }
};

File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.js
/**
 * @param {number[]} nums
 * @return {number}
 */
function solve(i, x, nums) {
    if(i == nums.length) return x;
    let pick = solve(i+1, x^nums[i], nums);
    let notPick = solve(i+1, x, nums);
    return pick + notPick;
}

var subsetXORSum = function(nums) {
    return solve(0, 0, nums);
};